src/server/req_runjob.c

```diff
@@ -265,6 +265,35 @@ clear_from_defr(int sd)
 
 /**
  * @brief
+ * 		wrapper function that calls process_hooks()
+ *
+ * 	@see
+ * 		req_runjob
+ *
+ * @return	int
+ * @retval	0 means at least one hook was encountered to have rejected the
+ *            request.
+ */
+int
+call_to_process_hooks(struct batch_request *preq, char *hook_msg, size_t msg_len,
+	void	(*pyinter_func)(void))
+{
+	switch(process_hooks(preq, hook_msg, msg_len, pbs_python_set_interrupt))
+	{
+		case 0:      /* explicit reject */
+				return 0;
+		case 1:    /* explicit accept */
+		case 2:    /* no hook script executed - go ahead and accept events*/
+				break;
+		default:
+				log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_HOOK,
+						LOG_INFO, "", "runjob event: accept req by default");
+	}
+
```

Comment:
You're falling out and not returning anything.  This will end up returning garbage.  As a note, if you compile your code with -Wall, the compiler would have warned you about this.  Of course compiling with -Wall is somewhat messy now, so it could easily be missed.